3.10/arangosearch-nested-search.md

```diff
@@ -0,0 +1,218 @@
+View definition like the following:
+
+```json
+{
+  "links": {
+    "<collection>": {
+      "fields": {
+        "dimensions": {
+          "fields": {
+            "type": {},
+            "value": {}
+          }
+        }
+      }
+    }
+  },
+  ...
+}
+```
+
+You might then write a query like the following to find documents where the
+height is greater than 40:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions.type == "height" AND doc.dimensions.value > 40
+  RETURN doc
+```
+
+This query matches the above document despite the height only being 35. The reason is
+that each condition is true for at least one of the nested objects. There is no
+check whether both conditions are true for the same object, however. You could
+add a `FILTER` statement to remove false positive matches from the search
+results, but it is cumbersome to check the conditions again, for every sub-object:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions.type == "height" AND doc.dimensions.value > 40
+  FILTER LENGTH(doc.dimensions[* FILTER CURRENT.type == "height" AND CURRENT.value > 40]) > 0
+  RETURN doc
+```
+
+The nested search feature allows you to condense the query while utilizing the
+View index:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions[? FILTER CURRENT.type == "height" AND CURRENT.value > 40]
+  RETURN doc
+```
+
+The required View definition for this to work is as follows:
+
+```json
+{
+  "links": {
+    "<collection>": {
+      "fields": {
+        "dimensions": {
+          "nested": {
+            "type": {},
+            "value": {}
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Note the `nested` property instead of a `fields` property, configuring the View
+to index the objects in the `dimensions` array so that you can use the
+[Question mark operator](aql/advanced-array-operators.html#question-mark-operator)
+to query the nested objects. The default `identity` Analyzer is used for the
+fields because none is specified explicitly.
+
+## Defining how often the conditions need to be true
+
+You can optionally specify a quantifier to define how often the conditions need
+to be true for the entire array. The following query matches documents that have
+one or two nested objects with a height greater than 40:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions[? 1..2 FILTER CURRENT.type == "height" AND CURRENT.value > 40]
+  RETURN doc
+```
+
+If you leave out the quantifier, it defaults to `ANY`. The conditions need to be
+fulfilled by at least one sub-object, but more than one sub-object may meet the
+conditions. With a quantity of `1`, it would need to be one match exactly.
+Similarly, ranges require an exact match between the minimum and maximum number,
+including the specified boundaries. To require two or more sub-objects to
+fulfill the conditions, you can use `AT LEAST (2)`, and so on.
+
+{% hint 'info' %}
+The `ALL` quantifier of the Question mark operator is not supported in View queries.
```

Comment:
by `View` here we mean `arangosearch` views, but this is supported for `search-alias` views. In general I believe we have to revisit documentation so we can make a distinction between `arangosearch` and `search-alias` explicit.

Reply:
Good point. Should we call them ArangoSearch Views and Search Alias Views, or rather something like Views of type `arangosearch` / `search-alias`?